lib/engine/game/g_steam_over_holland/game.rb

```diff
@@ -0,0 +1,235 @@
+
+        PHASES = [{ name: '2', train_limit: 4, tiles: [:yellow] },
+                  {
+                    name: '3',
+                    on: '3',
+                    train_limit: 4,
+                    tiles: %i[yellow green],
+                    status: ['can_buy_companies'],
+                  },
+                  {
+                    name: '4',
+                    on: '4',
+                    train_limit: 3,
+                    tiles: %i[yellow green],
+                    status: ['can_buy_companies'],
+                  },
+                  {
+                    name: '5',
+                    on: '5',
+                    train_limit: 2,
+                    tiles: %i[yellow green brown],
+                  },
+                  {
+                    name: '6',
+                    on: '6',
+                    train_limit: 2,
+                    tiles: %i[yellow green brown],
+                  }].freeze
+
+        TRAINS = [{ name: '2', distance: 2, price: 100, rusts_on: '4', num: 5 },
+                  { name: '3', distance: 3, price: 200, rusts_on: '5', num: 4 },
+                  { name: '4', distance: 4, price: 300, rusts_on: '6', num: 3 },
+                  {
+                    name: '5',
+                    distance: 3,
+                    price: 400,
+                    num: 3,
+                    events: [{ 'type' => 'close_companies' }],
+                  },
+                  {
+                    name: '6',
+                    distance: 6,
+                    price: 500,
+                    num: 6,
+                    variants: [
+                      {
+                        name: '3E',
+                        distance:
+                          [
+                            { 'nodes' => %w[city offboard], 'pay' => 3, 'visit' => 3, 'multiplier' => 2 },
+                            { 'nodes' => ['town'], 'pay' => 0, 'visit' => 99 },
+                          ],
+                        price: 600,
+                      },
+                    ],
+                  }].freeze
+
+        def timeline
+          @timeline ||= [
+            'Game ends after OR 5.2!',
+          ].freeze
+          @timeline
+        end
+
+        def show_progress_bar?
+          true
+        end
+
+        def progress_information
+          base_progress = [
+            { type: :PRE },
+            { type: :SR },
+            { type: :OR, name: '1.1' },
+            { type: :OR, name: '1.2' },
+            { type: :SR },
+            { type: :OR, name: '2.1' },
+            { type: :OR, name: '2.2' },
+            { type: :SR },
+            { type: :OR, name: '3.1' },
+            { type: :OR, name: '3.2' },
+            { type: :SR },
+            { type: :OR, name: '4.1' },
+            { type: :OR, name: '4.2' },
+            { type: :SR },
+            { type: :OR, name: '5.1' },
+            { type: :OR, name: '5.2' },
+          ]
+
+          base_progress << { type: :End }
+        end
```

Comment:
You don't need to be creating and modifying an array here.
```suggestion
        def progress_information
          [
            { type: :PRE },
            { type: :SR },
            { type: :OR, name: '1.1' },
            { type: :OR, name: '1.2' },
            { type: :SR },
            { type: :OR, name: '2.1' },
            { type: :OR, name: '2.2' },
            { type: :SR },
            { type: :OR, name: '3.1' },
            { type: :OR, name: '3.2' },
            { type: :SR },
            { type: :OR, name: '4.1' },
            { type: :OR, name: '4.2' },
            { type: :SR },
            { type: :OR, name: '5.1' },
            { type: :OR, name: '5.2' },
            { type: :End },
          ]
        end
```

Reply:
Thanks, I stole this idea wholesale from 18MS.